src/steamship/agents/context.py

```diff
@@ -0,0 +1,39 @@
+from typing import Dict, Any, List, Optional
+
+from pydantic import BaseModel
+
+from steamship import Steamship
+from steamship.experimental.transports.chat import ChatMessage
+
+
+class ChatHistory(BaseModel):
+    chat_id: str = ""  # chat history is scoped by a chat_id. chat_id is unique across user_id's
+    messages: List[str] = []  # Represents a database that can select & sort messages
+
+    def get_history(self, message: ChatMessage, context):
+        return self.messages
+
+    def add_user_message(self, message: ChatMessage):
+        self.messages.append(f"user: {message}")
+
+    def add_ai_message(self, message: ChatMessage):
+        self.messages.append(f"AI: {message}")
+
+    def clear(self):
+        self.messages = []
+
+
+class AgentContext(BaseModel):
+    client: Steamship
+    chat_history: Optional[ChatHistory] = ChatHistory()
+    tracing: Optional[Dict[str, Any]] = []
+    scratchpad: Optional[List[str]] = []
+    new_message: ChatMessage
```

Comment:
Do we want a vector store here?  Is it separate from ChatHistory, or a different form thereof?

Reply:
Yea, that's why I've included ChatHistory so we could have the flexibility to add a vector store in there. Not sure if ChatHistory contains a vector store by default, or if that's a subclass of ChatHistory.